### ISSUE TITLE
Tabs doesn't handle capitalized data-id values

### ISSUE THREAD

**Issue by NicholasBoll** [Workday/canvas-kit#2354]
## 🐛 Bug Report

The `Tabs` component does not handle capitalized identifiers. The `Tabs.Panel` has an `id` attribute that uses `slugify` while the `Tabs.Item` component has an `aria-controls` that does not use `slugify`.

- `Tabs.Panel`: https://github.com/Workday/canvas-kit/blob/edea2541425f275aeb8ada02b4c84eddf3a5b96a/modules/react/tabs/lib/TabsPanel.tsx#L60
- `Tabs.Item`: https://github.com/Workday/canvas-kit/blob/edea2541425f275aeb8ada02b4c84eddf3a5b96a/modules/react/tabs/lib/TabsItem.tsx#L175

The fix would be to use `slugify` in `Tabs.Item` on the `aria-controls` attribute.

## To Reproduce

- Visit https://codesandbox.io/s/named-tabs-capital-wzcv88?file=/src/App.tsx
- Inspect the DOM of the tab item `[role=tab]` and note the `aria-controls` and the `[role=tabpanel]` and note the `id`

## Expected Behavior

The `id` of the tab panel and the `aria-controls` of the tab item should be the same

## Actual Results

```html
<!-- [role=tab] note the `aria-controls`  has a capital "First" -->
<button type="button" role="tab" aria-selected="true" aria-controls="tabpanel-y1me0-First" data-focus-id="y1me0-First" tabindex="0" data-id="First" id="y1me0-first" class="css-20dw3s" style="">First Tab</button>

<!-- [role=tabpanel] not the `id` has a lowercase "first" -->
<div role="tabpanel" aria-labelledby="y1me0-first" id="tabpanel-y1me0-first" tabindex="0" data-id="First" class="css-hmwn8z">Contents of First Tab</div>
```
